app/services/punches_service.rb

```diff
@@ -0,0 +1,15 @@
+class PunchesService
+  def self.create(punches, user)
+    deletes = punches.map { |punch| punch['from'] }
+
+    @punches = user.punches
+    created_punches = @punches.transaction do
+      @punches.by_days(deletes).delete_all
+      @punches.where(
+        company: user.company
+      ).create(punches)
+    end
+
+    return created_punches
```

Comment:
```suggestion
```
Since this value is already returned by the previous block, you don't need this explicit return.